addon/views/header-block.js

```diff
@@ -9,5 +9,12 @@ export default TableBlock.extend({
 
   content: Ember.computed(function() {
     return [this.get('columns')];
-  }).property('columns')
+  }).property('columns'),
+
+  onColumnsDidChange: Ember.observer(function() {
+    var _this = this;
+    Ember.run.schedule('afterRender', function() {
```

Comment:
Not 100% sure but pretty sure this will work without, since it's always in the same run loop and should happen before a `destroy`